kstreams/streams.py

```diff
@@ -129,3 +143,26 @@ async def __anext__(self) -> structs.ConsumerRecord:
             return consumer_record
         except errors.ConsumerStoppedError:
             raise StopAsyncIteration  # noqa: F821
+
+
+# Function required by the `stream` decorator
+Func = Callable[[Stream], Awaitable[Any]]
```

Comment:
This should be `Func = Callable[[Stream], None]` because the `coroutine/ async generator` that receives a `stream` as argument doesn't return anything. Also, I would rename it because `Func` is generic, maybe something like `StreamFunc`

Reply:
It cannot be `None` because mypy fails. But I can make it `Awaitable[None]` instead.

The thing is, if I do that, all streams should have the signature `-> None`. Because a function by default returns `Any`.

So this would not match `@stream` signature if using `None`:

```py
@stream(...)
def consume(stream: Stream):
      ....
```

That's why I change it to `Any`. What do you think we should do then?

Reply:
When a `stream` is defined, it always return `None` or it  could `yield` something. 

```python
@stream(...)
async def my_stream(stream: Stream):
   ...
```

Maybe with the new changes I can not see it properly, but `my_stream` (Func) is a callable object that receives a `stream` and it always returns `None`.

Reply:
I understand what you are saying, but in python code, if you don't specify the return signature, it defaults to `Any`.
So this: `async def my_stream(stream: Stream):` is the same as `async def my_stream(stream: Stream) -> Any:`

That's why I used `Any`. If we want to use `None`, then in order to not see linting errors, users will have to manually say this: `async def my_stream(stream: Stream) -> None:`